sentry_sdk/integrations/launchdarkly.py

```diff
@@ -0,0 +1,73 @@
+from typing import TYPE_CHECKING
+import sentry_sdk
+
+from sentry_sdk.integrations import DidNotEnable, Integration
+
+try:
+    import ldclient
+    from ldclient.hook import Hook, Metadata
+
+    if TYPE_CHECKING:
+        from ldclient import LDClient
+        from ldclient.hook import EvaluationSeriesContext
+        from ldclient.evaluation import EvaluationDetail
+
+        from sentry_sdk._types import Event, ExcInfo
+        from typing import Any, Optional
+except ImportError:
+    raise DidNotEnable("LaunchDarkly is not installed")
+
+
+class LaunchDarklyIntegration(Integration):
+    identifier = "launchdarkly"
+
+    def __init__(self, client=None):
+        # type: (LDClient | None) -> None
+        """
+        @param client    An initialized LDClient instance. If a client is not provided, this integration will attempt to
+                         use the shared global instance. This will fail if ldclient.set_config() hasn't been called.
+
+        Docs reference: https://docs.launchdarkly.com/sdk/server-side/python
+        """
+        if client is None:
+            try:
+                client = ldclient.get()  # global singleton.
+            except Exception as exc:
+                raise DidNotEnable("Error getting LaunchDarkly client. " + repr(exc))
+
+        if not client.is_initialized():
+            raise DidNotEnable("LaunchDarkly client is not initialized.")
+
+        # Register the flag collection hook with the given client.
+        client.add_hook(LaunchDarklyHook())
```

Comment:
This needs to be moved to setup_once.  You can call `sentry_sdk.get_client().get_integration(LaunchDarklyIntegration)` to get the explicitly passed client from the init.

Reply:
`client` here is the launchdarkly client.

Reply:
Got it, when is `setup_once` called? Do you know when setup_once is called? Why is it a staticmethod?